### PR TITLE
[collectd 6] interface plugin: Align metrics with OpenTelemetry recommendations.

### PR DESCRIPTION
**Pull request by octo**
https://opentelemetry.io/docs/specs/semconv/system/system-metrics/#network-metrics

ChangeLog: Interface plugin: The metric schema has been aligned with OpenTelemetry semantic conventions.

### CODE REVIEW

File: src/interface.c [hunk 1]
```diff
@@ -97,47 +97,19 @@ static size_t numif;
...
-    .name = "node_network_transmit_errs_total",
-    .help = "Network device statistic transmit_errs.",
-    .type = METRIC_TYPE_COUNTER,
-};
-metric_family_t transmit_packets = {
-    .name = "node_network_transmit_packets_total",
-    .help = "Network device statistic transmit_packets.",
-    .type = METRIC_TYPE_COUNTER,
-};
+static char const *const direction_label = "network.io.direction";
```
Comment (eero-t) [inline src/interface.c:108]: Could have link to spec as comment.

**Review by eero-t — COMMENTED** [review]
Code looks OK, but IMHO there should be some migration doc where all these metric name/label changes from v5 are documented, at the same time they are changed in the plugins.

I don't think anybody's going to go through all the plugins to document it, and catch all the changes afterwards, at release time.

But such documentation is absolutely necessary because users are going to need to update metric names in all their collectd integrations (Prometheus monitors, Grafana dashboards, SQL injection scripts etc)!

File: src/interface.c (hunk 1 above)
Comment (octo) [reply src/interface.c:110]: Done.

**Comment by octo** [comment]
> there should be some migration doc where all these metric name/label changes from v5 are documented, at the same time they are changed in the plugins.

I started to document this here: https://github.com/collectd/collectd/wiki/collectd-6

**Comment by eero-t** [comment]
> I started to document this here: https://github.com/collectd/collectd/wiki/collectd-6

Great!    (Noticed typo there:  /ReportUtilizatoin//ReportUtilization/)

**Review by eero-t — APPROVED** [review]
On quick review, looks OK.

**Comment by octo** [comment]
> Noticed typo there: /ReportUtilizatoin/ReportUtilization/

Thanks for flagging this! It's a wiki. You should also have access to fix it.

**Pull request merged**